### PR TITLE
NU5 testnet update

### PR DESCRIPTION
**Pull request by str4d**

### CODE REVIEW

**Comment by defuse** [comment]
utACK I double-checked hashes and constants

File: zcash_primitives/src/consensus.rs [hunk 1]
```diff
@@ -240,7 +240,7 @@ impl Parameters for TestNetwork {
             NetworkUpgrade::Blossom => Some(BlockHeight(584_000)),
             NetworkUpgrade::Heartwood => Some(BlockHeight(903_800)),
             NetworkUpgrade::Canopy => Some(BlockHeight(1_028_500)),
-            NetworkUpgrade::Nu5 => None,
+            NetworkUpgrade::Nu5 => Some(BlockHeight(1_599_200)),
```
Comment (str4d) [inline zcash_primitives/src/consensus.rs:243]: I calculate this to be 8:28pm UTC, October 7th.

File: zcash_primitives/src/consensus.rs [hunk 2]
```diff
@@ -458,7 +458,7 @@ impl TryFrom<u32> for BranchId {
             0x2bb4_0e60 => Ok(BranchId::Blossom),
             0xf5b9_230b => Ok(BranchId::Heartwood),
             0xe9ff_75a6 => Ok(BranchId::Canopy),
-            0xf919_a198 => Ok(BranchId::Nu5),
+            0x3751_9621 => Ok(BranchId::Nu5),
```
Comment (str4d) [inline zcash_primitives/src/consensus.rs:461]: Thanks `/dev/urandom` for your contribution 😄

File: zcash_primitives/src/consensus.rs (hunk 1 above)
Comment (defuse) [reply zcash_primitives/src/consensus.rs:243]: Did we miss this change in the previous release?

**Comment by str4d** [comment]
ZIP 244 test vectors are updated using https://github.com/zcash-hackworks/zcash-test-vectors/pull/54.

**Comment by defuse** [comment]
utACK again :)

File: zcash_primitives/src/consensus.rs (hunk 1 above)
Comment (therealyingtong) [reply zcash_primitives/src/consensus.rs:243]: I've double-checked this calculation.

File: Cargo.toml [hunk 3]
```diff
@@ -20,8 +20,7 @@ codegen-units = 1
 
 [patch.crates-io]
 # In development.
-halo2 = { git = "https://github.com/zcash/halo2.git", rev = "a7cd600eb60b1528159b92af5e426adcc615de1a" }
-orchard = { git = "https://github.com/zcash/orchard.git", rev = "8779ce8f1a638ebbc9b229d4eff3a29ef4de7ac0" }
+orchard = { git = "https://github.com/zcash/orchard.git", rev = "2c8241f25b943aa05203eacf9905db117c69bd29" }
```
Comment (therealyingtong) [inline Cargo.toml:23]: I've checked that this commit merges the circuit bugfixes in https://github.com/zcash/orchard/pull/209.

**Review by nuttycom — APPROVED** [review]
utACK

**Review by therealyingtong — APPROVED** [review]
utACK

File: zcash_primitives/src/consensus.rs (hunk 1 above)
Comment (daira) [reply zcash_primitives/src/consensus.rs:243]: @defuse yes we did.

**Review by daira — COMMENTED** [review]
Post-hoc utACK

File: zcash_primitives/src/consensus.rs (hunk 1 above)
Comment (str4d) [reply zcash_primitives/src/consensus.rs:243]: It wasn't technically necessary, as we don't use these parameters in zcashd. But it's good practice to update them in sync.

**Pull request merged**